src/VecSim/algorithms/hnsw/hnsw_factory.cpp

```diff
@@ -73,3 +71,13 @@ size_t HNSWFactory::EstimateElementSize(const HNSWParams *params) {
      */
     return size_meta_data + size_total_data_per_element;
 }
+
+// TODO overload for doubles
+VecSimBatchIterator *newBatchIterator(void *queryBlob, VecSimQueryParams *queryParams,
+                                      std::shared_ptr<VecSimAllocator> allocator,
+                                      HNSWIndex<float, float> *index) {
```

Comment:
index type?

Reply:
this function will be overloaded
VecSimBatchIterator *newBatchIterator(void *queryBlob, VecSimQueryParams *queryParams,
                                      std::shared_ptr<VecSimAllocator> allocator,
                                      HNSWIndex<double, double> *index)